qiskit_machine_learning/algorithms/classifiers/qsvc.py

```diff
@@ -66,5 +74,54 @@ def quantum_kernel(self) -> QuantumKernel:
     @quantum_kernel.setter
     def quantum_kernel(self, quantum_kernel: QuantumKernel):
         """Sets quantum kernel"""
-        self._quantum_kernel = quantum_kernel
+        self._kernel_trainer = None
+
+        # If no quantum kernel passed, generate a default QuantumKernel
+        if not quantum_kernel:
+            backend = Aer.get_backend("qasm_simulator")
+            self._quantum_kernel = QuantumKernel(quantum_instance=backend)
+
+        # QuantumKernel passed
+        elif isinstance(quantum_kernel, QuantumKernel):
+            self._quantum_kernel = quantum_kernel
+
+            # If there are unbound user parameters, instantiate a QKTrainer
+            unbound_params = quantum_kernel.unbound_user_parameters()
+            if unbound_params:
+                # If user has not specified they wish to train the kernel,
+                # provide a warning
+                if not self.train_kernel:
+                    warnings.warn(
+                        f"""
+                        Unbound user parameters were detected in the quantum
+                        kernel feature map ({unbound_params}). If these are
+                        not bound before calling QSVC.fit, a computationally
+                        expensive kernel training procedure will precede
+                        fitting the SVC.
+                        """,
+                        UserWarning,
+                    )
+
+                self._kernel_trainer = QuantumKernelTrainer(quantum_kernel)
+
+        # If a QKTrainer was passed, set the quantum_kernel and kernel trainer fields
+        elif isinstance(quantum_kernel, QuantumKernelTrainer):
+            self._quantum_kernel = quantum_kernel.quantum_kernel
+            self._kernel_trainer = quantum_kernel
         self.kernel = self._quantum_kernel.evaluate
+
+    @property
+    def kernel_trainer(self) -> QuantumKernelTrainer:
+        """Returns quantum kernel trainer"""
+        return self._kernel_trainer
+
+    def fit(self, X: np.ndarray, y: np.ndarray, sample_weight=None):
+        """
+        Wrapper method for SVC.fit which optimizes the quantum kernel's
+        user parameters before fitting the SVC.
```

Comment:
This should document args since this will now come out in docs. It subclasses the method, and it maybe enough to say that if a kernel trainer instance was given instead of a kernel, then the kernel will first be trained.